posthog/warehouse/models/join.py

```diff
@@ -44,9 +45,7 @@ def join_function(
         self, override_source_table_key: Optional[str] = None, override_joining_table_key: Optional[str] = None
     ):
         def _join_function(
-            from_table: str,
-            to_table: str,
-            requested_fields: dict[str, Any],
+            join_to_add: LazyJoinToAdd,
```

Comment:
This is much nicer, trying to find all references to a `join_function` before was quite hard. Having something typed like this makes it a degree easier 👏